weave/utils/retry.py

```diff
@@ -14,13 +16,18 @@
 
 T = TypeVar("T")
 
+# Context variable to store retry ID for correlation
+_retry_id: contextvars.ContextVar[str] = contextvars.ContextVar("retry_id")
```

Comment:
There's only 1 var here -- can only 1 thing be retried at a time per thread?

Reply:
yeah... but I think that each thread should always only be doing one thing right? 1 thread per conn. I think...